include/fiction/algorithms/simulation/sidb/assess_physical_popstability.hpp

```diff
@@ -0,0 +1,160 @@
+//
+// Created by Jan Drewniok on 02.11.23.
+//
+
+#ifndef FICTION_QUANTITY_STABILITY_OF_GROUND_STATE_HPP
+#define FICTION_QUANTITY_STABILITY_OF_GROUND_STATE_HPP
+
+#include "fiction/algorithms/simulation/sidb/quickexact.hpp"
+#include "fiction/algorithms/simulation/sidb/sidb_simulation_parameters.hpp"
+#include "fiction/layouts/cell_level_layout.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
+
+#include <cmath>
+#include <limits>
+#include <tuple>
+#include <unordered_map>
+#include <utility>
+
+namespace fiction
+{
+
+/**
+ * This defines the various types of charge transitions that can occur in a SiDB layout. These transitions represent
+ * changes in the charge state of SiDBs, including transitions from neutral to negative, negative to neutral, neutral to
+ * positive, and positive to neutral.
+ */
+enum class transition_type
+{
+    NEUTRAL_TO_NEGATIVE,
+    NEGATIVE_TO_NEUTRAL,
+    NEUTRAL_TO_POSITIVE,
+    POSITIVE_TO_NEUTRAL,
+};
+
+/**
+ * This function calculates the minimum electrostatic potential to the closest charge transition for a SiDB layout at
+ * given physical parameters. It records information about the SiDB which is closest to a charge transition, the
+ * transition type, and the electrostatic potential needed for the charge transition.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout for which pop stability is assessed.
+ * @param params The simulation parameters used for the assessment.
+ * @return An unordered map where each key represents the charge distribution in ascending energy order (i.e., key = 0
+ * is the ground state of the system), and the value is a tuple containing the SiDB which is closest to a charge
```

Comment:
What good does it do me if I get told information about the n-th charge distribution, but have no idea what it looks like? Why would you discard these distributions instead of returning them, too, instead of just the indexes? You compute them anyway, so you might as well return them. Then, the caller can decide what to do with them.